lib/archethic/networking/ip_lookup/public_gateway.ex

```diff
@@ -0,0 +1,9 @@
+defmodule Archethic.Networking.IPLookup.PublicGateway do
+  @moduledoc false
+
+  use Knigge,
+    otp_app: :archethic,
+    default: Archethic.Networking.IPLookup.IPIFY
+
+  @callback get_node_ip() :: {:ok, :inet.ip_address()} | {:error, :not_recognizable_ip}
```

Comment:
I think you can specify the `behavior` in Knigge, to avoid duplication of the callbacks.
Hence this behaviour will be chained to `IPLookup.Impl` behaviour , and Knigge will provide abstraction for it. (https://github.com/sascha-wolf/knigge#the-behaviour-key-is-optional)

Reply:
marking it as resolved ,
as we have new implementation